models/autocomplete_model.py

```diff
@@ -0,0 +1,34 @@
+from pathlib import Path
+import os
+import re
+
+import discord
+from models.usage_service_model import UsageService
+from models.openai_model import Model
+
+usage_service = UsageService(Path(os.environ.get("DATA_DIR", os.getcwd())))
+model = Model(usage_service)
+
+
+class Settings_autocompleter:  
+    async def get_settings(ctx: discord.AutocompleteContext):
+        SETTINGS = [re.sub("^_","",key) for key in model.__dict__.keys() if key not in model._hidden_attributes]
+        return [parameter for parameter in SETTINGS if parameter.startswith(ctx.value.lower())][:25]
+    async def get_value(ctx: discord.AutocompleteContext): # Behaves a bit weird if you go back and edit the parameter without typing in a new command
+        values = {
+            'mode' : ['temperature', 'top_p'],
+            'model' : ["text-davinci-003", "text-curie-001"],
+            'low_usage_mode' : ["True", "False"],
+            'image_size' : ["256x256", "512x512", "1024x1024"],
+            'summarize_conversastion' : ["True", "False"],
+            'welcome_message_enabled' : ["True", "False"]
+        }
+        if ctx.options["parameter"] in values.keys():
+            return[value for value in values[ctx.options["parameter"]]]
+        else:
+            await ctx.interaction.response.defer() # defer so the autocomplete in int values doesn't error but rather just says not found
+            return []
+
+class File_autocompleter:
+    async def get_openers(ctx: discord.AutocompleteContext):
+        return [file for file in os.listdir('openers') if file.startswith(ctx.value.lower())][:25] # returns the 25 first files from your current input
```

Comment:
Can we add some error handling to this? For example if the openers directory is accidentally deleted by the user or something, maybe we should pop an error and say that there is no `openers` directory.

(when autocomplete happens with no openers directory)
```
future: <Task finished name='Task-36' coro=<ApplicationCommandMixin.on_application_command_auto_complete.<locals>.callback() done, defined at /usr/local/lib/python3.9/dist-packages/discord/bot.py:853> exception=FileNotFoundError(2, 'No such file or directory')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/discord/bot.py", line 856, in callback
    return await command.invoke_autocomplete_callback(ctx)
  File "/usr/local/lib/python3.9/dist-packages/discord/commands/core.py", line 1316, in invoke_autocomplete_callback
    await command.invoke_autocomplete_callback(ctx)
  File "/usr/local/lib/python3.9/dist-packages/discord/commands/core.py", line 1003, in invoke_autocomplete_callback
    result = await result
  File "/home/gptbot/models/autocomplete_model.py", line 34, in get_openers
    return [file for file in os.listdir('openers') if file.startswith(ctx.value.lower())][:25] # returns the 25 first files from your current input
FileNotFoundError: [Errno 2] No such file or directory: 'openers'
```

Reply:
Yeah no problem